### ISSUE TITLE
Potential issue with array

### ISSUE THREAD

**Issue by hea-lab** [jruizgit/rules#287]
From my understanding, the following rule should trigger the associated action. But it does not with the current code.

Is this allowed by the current framework?

```
d.ruleset('test', function() {
        whenAll: m.array.anyItem(item.field2 == m.field1)
        run: console.log("ok", JSON.stringify(m))
});

d.post('test',{field1: 8, array: [{field2 : 8}]});
```

**Comment by jruizgit** [comment]
Thanks for reporting this issue. This case should work. I have pushed a fix. Please use version 2.0.20

**Comment by hea-lab** [comment]
your fix works properly, thanks
